<syscall.h> shouldn't require malloc

When including `syscall.h` after making `malloc` unavailable an error is thrown.

### Why do you need this?

I would like to make progress adding serenity support to a a system call fuzzer named syzkaller (https://github.com/SerenityOS/serenity/issues/1025). I have a POC, [but it requires a few hacks that aren't upstreamable](https://github.com/HerrSpace/syzkaller/commit/799fbf7fc9ce3c034e77cbd1a9b2fd98f05c8a50).

[Syzkaller deliberately makes malloc unavailable to enhance test reproducibility](https://github.com/google/syzkaller/blob/master/executor/executor.cc#L59-L64). As this works fine for every other operating system supported by syzkaller, I assume serenity should not require malloc here either.

### Minimal Repro

dummy.cpp:
```cpp
#include <stdlib.h>
#define malloc do_not_use_malloc
#include <syscall.h>
```

### Compiler Output
```
$ ~/repos/upstream/serenity/Toolchain/Local/i686/bin/i686-pc-serenity-gcc -I ~/repos/upstream/serenity/Build/i686/Root/usr/include/ -std=c++20 -o ./dummy dummy.cpp

/Users/space/repos/upstream/serenity/Build/i686/Root/usr/include/AK/kmalloc.h: In function 'void* kmalloc_array(AK::Checked<unsigned int>, AK::Checked<unsigned int>)':
dummy.cpp:2:16: error: 'do_not_use_malloc' was not declared in this scope
    2 | #define malloc do_not_use_malloc
      |                ^~~~~~~~~~~~~~~~~
/Users/space/repos/upstream/serenity/Build/i686/Root/usr/include/AK/kmalloc.h: In function 'void* kmalloc_array(AK::Checked<unsigned int>, AK::Checked<unsigned int>, AK::Checked<unsigned int>)':
dummy.cpp:2:16: error: 'do_not_use_malloc' was not declared in this scope
    2 | #define malloc do_not_use_malloc
      |                ^~~~~~~~~~~~~~~~~
```

--- a/Kernel/API/Syscall.h
+++ b/Kernel/API/Syscall.h
@@ -6,7 +6,6 @@
 
 #pragma once
 
-#include <AK/StringView.h>
 #include <AK/Types.h>
 #include <AK/Userspace.h>
 
@@ -200,21 +199,6 @@ enum Function {
 #undef __ENUMERATE_SYSCALL
         __Count
 };
-
-constexpr StringView to_string(Function function)
-{
-    switch (function) {
-#undef __ENUMERATE_SYSCALL
-#define __ENUMERATE_SYSCALL(sys_call, needs_lock) \
-    case SC_##sys_call:                           \
-        return #sys_call##sv;
-        ENUMERATE_SYSCALLS(__ENUMERATE_SYSCALL)
-#undef __ENUMERATE_SYSCALL
-    default:
-        break;
-    }
-    return "Unknown"sv;
-}
 
 #ifdef __serenity__
 struct StringArgument {

--- a/Kernel/API/SyscallString.h
+++ b/Kernel/API/SyscallString.h
@@ -1,0 +1,32 @@
+/*
+ * Copyright (c) 2018-2022, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, Patrick Meyer <git@the-space.agency>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StringView.h>
+#include <Kernel/API/Syscall.h>
+
+namespace Kernel::Syscall {
+
+// Separate header so syscall.h doesn't depend on malloc.
+// https://github.com/SerenityOS/serenity/issues/13869
+constexpr StringView to_string(Function function)
+{
+    switch (function) {
+#undef __ENUMERATE_SYSCALL
+#define __ENUMERATE_SYSCALL(sys_call, needs_lock) \
+    case SC_##sys_call:                           \
+        return #sys_call##sv;
+        ENUMERATE_SYSCALLS(__ENUMERATE_SYSCALL)
+#undef __ENUMERATE_SYSCALL
+    default:
+        break;
+    }
+    return "Unknown"sv;
+}
+
+}

--- a/Tests/Kernel/fuzz-syscalls.cpp
+++ b/Tests/Kernel/fuzz-syscalls.cpp
@@ -9,6 +9,7 @@
 #include <AK/String.h>
 #include <AK/StringBuilder.h>
 #include <AK/Vector.h>
+#include <Kernel/API/SyscallString.h>
 #include <errno.h>
 #include <signal.h>
 #include <stdio.h>

--- a/Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp
+++ b/Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp
@@ -12,6 +12,7 @@
 #include <AK/Debug.h>
 #include <AK/FileStream.h>
 #include <AK/Format.h>
+#include <Kernel/API/SyscallString.h>
 #include <alloca.h>
 #include <fcntl.h>
 #include <sched.h>

--- a/Userland/Utilities/functrace.cpp
+++ b/Userland/Utilities/functrace.cpp
@@ -8,6 +8,7 @@
 #include <AK/HashMap.h>
 #include <AK/NonnullOwnPtr.h>
 #include <AK/StringBuilder.h>
+#include <Kernel/API/SyscallString.h>
 #include <LibC/sys/arch/i386/regs.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/File.h>

--- a/Userland/Utilities/strace.cpp
+++ b/Userland/Utilities/strace.cpp
@@ -9,6 +9,7 @@
 #include <AK/IPv4Address.h>
 #include <AK/StdLibExtras.h>
 #include <AK/Types.h>
+#include <Kernel/API/SyscallString.h>
 #include <LibC/sys/arch/i386/regs.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/File.h>

--- a/Userland/Utilities/syscall.cpp
+++ b/Userland/Utilities/syscall.cpp
@@ -9,6 +9,7 @@
 #include <AK/Debug.h>
 #include <AK/Iterator.h>
 #include <AK/Vector.h>
+#include <Kernel/API/SyscallString.h>
 #include <LibCore/ArgsParser.h>
 #include <LibMain/Main.h>
 #include <errno_codes.h>

